ast/ast.go

```diff
@@ -182,7 +182,7 @@ type Statement interface {
 	OriginText() string
 
 	// Exec executes SQL and gets a Recordset.
-	Exec(ctx context.Context) (RecordSet, error)
+	Exec(goCtx goctx.Context, ctx context.Context) (RecordSet, error)
```

Comment:
Seems we don't need the second `context.Context`.
The `Statement` has one in its struct.